scripts/match_annexscientificname_to_scientificname.py

```diff
@@ -0,0 +1,127 @@
+from helpers import get_database_connection, get_config, setup_log_file, execute_sql_from_jinja_string, execute_sql_from_file
+import time
+import logging
+
+
+def _insert_or_get_scientificname(conn, scientific_name, authorship):
+    """ Insert or select a name in scientificname table based on its scientific name and authorship
+
+        If the scientific name - authorship combination already exists in the scientificname table, select it.
+        Otherwise, insert it in a new row.
+
+        In both cases, returns the row id """
+
+    sc_name_template = """WITH ins AS (
+        INSERT INTO scientificname ("scientificName", "authorship")
+        VALUES ({{ scientific_name }}, {{ authorship}})         -- input value
+        ON CONFLICT DO NOTHING
+        RETURNING scientificname.id
+        )
+    SELECT id FROM ins
+    UNION  ALL
+    SELECT "id" FROM scientificname          -- 2nd SELECT never executed if INSERT successful
+    {% if authorship is defined %}
+        WHERE "scientificName" = {{ scientific_name }} AND "authorship" = {{ authorship }} -- input value a 2nd time
+    {% else %}
+        WHERE "scientificName" = {{ scientific_name }} AND "authorship" is NULL -- input value a 2nd time
+    {% endif %}
+    LIMIT  1;"""
+    cur = execute_sql_from_jinja_string(conn,
+                                        sql_string=sc_name_template,
+                                        context={'scientific_name': scientific_name,
+                                                 'authorship': authorship},
+                                        dict_cursor=True)
+    return cur.fetchone()['id']
+
+def _update_scientificname_id(conn, scientificname_id, row_id):
+    """ Add scientificNameId in annexscientificname if the name is found in scientificname table"""
+    template = """ UPDATE annexscientificname SET "scientificNameId" = {{ scientificname_id }} """\
+               """ WHERE "id" = {{ id }} """
+    execute_sql_from_jinja_string(conn,
+                                  template,
+                                  {'scientificname_id': scientificname_id,
+                                   'id': row_id})
+
+def _remove_aux_columns(conn):
```

Comment:
Maybe I'm missing something, but it seems a bit weird to create to columns in `create_new_tables` to finally remove them here. Are they temporary necessary in the meantime? Otherwise I'd suggest to not create them initially, and to remove this function.

Reply:
I understand your point.

Yes, they are temporary necessary. 
Not using them, it means checking the presence of this names in `scientificname` on the flow while populating the table `annexscientificname` instead of populating the table first and then matching to `scientificname`.
I thought to split the two steps for making things cleaner but removing columns afterwards is not such clean.

Reply:
I am refactoring by first adding the corrected version of the name (`scientificName`, `authorship`) from csv file to scientificname table, with the SQL INSERT-SELECT statement and then adding the other fields from file to anenxscientificname. As extra benefit we have a function less, `_update_scientificname_id()` as we don't have to update `scientificNameId` in db as we insert it in `annexscientificname` together with the other fields.

Reply:
I have just realized why I opted to maintain the match to scientificname as a separate step: if they add some new annexes in database the match step can be performed autonomously. What do you think, @niconoe? I found it a good argument while implementing it.

Reply:
Actually, I have to correct myself. As we removed the two columns in `match_annex_scientificname_to_scientificname()`, it was not reusable in maintenance phase as well. So, refactoring as we are doing is surely good. I will think to incapsulate code which will be useful for maintenance in a function so that it can be reusable later.

Reply:
Yep, I guess having a single step at migration time (that relies on reusable functions that we can call from maintenance scripts) seems like the best approach for now. Do you agree?